src/intTest/resources/fhir/screening.fhir.json

```diff
@@ -0,0 +1,162 @@
+{
+  "resourceType": "Bundle",
+  "id": "bfbbff2b-aa31-455e-8072-43ef3f28a13a",
+  "meta": {
+    "lastUpdated": "2021-03-12T13:36:29.588+00:00",
+    "profile": [ "https://fhir.nhs.uk/STU3/StructureDefinition/ITK-Message-Bundle-1" ]
+  },
+  "identifier": {
+    "system": "https://tools.ietf.org/html/rfc4122",
+    "value": "7ff987dc-eb21-43f3-9ec6-f9967feabf15"
+  },
+  "type": "message",
+  "entry": [
+    {
+      "fullUrl": "urn:uuid:68f3ed2a-9f5f-4d2d-84a9-0c4aaa5529de",
+      "resource": {
+        "resourceType": "Patient",
+        "id": "68f3ed2a-9f5f-4d2d-84a9-0c4aaa5529de",
+        "identifier": [
+          {
+            "system": "https://fhir.nhs.uk/Id/nhs-number",
+            "value": "8982821201"
+          }
+        ],
+        "name": [
+          {
+            "family": "GRIMSHAW",
+            "given": [ "BRETT" ],
+            "prefix": [ "CAPT" ]
+          }
+        ],
+        "gender": "male",
+        "birthDate": "1938-02-06",
+        "address": [
+          {
+            "postalCode": "TQ14 4QJ"
+          }
+        ]
```

Comment:
The `screening.edifact.dat` file seems to have address lines for the patient in the S06 segment, but they don't seem to be appearing here or in the adaptor translation. Not sure why this is happening

Reply:
I noticed that, seems to be an issue with the `UnstructuredAddress` parser here, https://github.com/nhsconnect/integration-adaptor-lab-results/blob/main/src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java#L38.

In our `screening.edifact.dat` example, the address lines length is 4, but should be greater than 5 according to the logic here (not sure why this is), so then `addressLines` is null

Reply:
That's to avoid NPEs. IIRC in the examples I'd been looking at, the empty lines of an address were indicated by an empty colon-delimited string. E.g., an address with only one line would be like `My house::::`, so I may have incorrectly assumed that was always the case. If those colon separators are only used when a subsequent line of the address is actually present, then `UnstructuredAddress` will need to be reworked

Reply:
As discussed in Teams, trailing colons **can** be omitted, so the `UnstructuredAddress` implementation incorrectly fails to parse messages without them.

Reply:
Since the "Cancer Screening - eGP-Comms GP Software Supplier Requirements v4.1" doc attached to https://gpitbjss.atlassian.net/browse/NIAD-998 doesn't specify any changes to the `ADR` segment, we have to go with what the NHS003 EDIFACT spec says, which means `ADR++US:HIGHWOOD BUNGALOW:EXETER:DEVON++TQ14 4QJ` in our example isn't valid. May just be an invalid example we've been sent?